pwndbg/lib/gcc.py

```
@@ -28,7 +28,7 @@ def which(arch: ArchDefinition) -> List[str]:
         if arch.ptrsize == 32:
             return ["g++", "-m32"]
         elif arch.ptrsize == 64:
-            return ["g++", "-m32"]
+            return ["g++", "-m64"]
         else:
             raise ValueError(f"Unknown pointer size: {arch.ptrsize}")
```

I believe this part was not correct even prior to the change. The exception gets triggered when I manually set `gcc = None`. The reason being `arch.ptrsize=8` instead of 64.

---

However, in the case that `g++` cannot be found with `_which_binutils`, I'm not sure why it doesn't just immediately throws an exception instead of having a fallback. But otherwise looks correct to me just that the fallback path hasn't been triggered for me so far as `g++` is installed for me.

---

Oop - yeah, `.ptrsize` is in bytes instead of bits (`.ptrbits` for bits).

---

> The reason being arch.ptrsize=8 instead of 64.

oh wow true, I tested it and it seemed to me like my patch changed something but I guess I hallucinated xd